### ISSUE TITLE
Please move to GitLab (Freedesktop or GNOME)

### ISSUE THREAD

**Issue by ghost** [flathub/flathub#419]
I want to close my GitHub account since months but I'm blocked by flathub.

With recent GitHub news, I think it's time to open an issue.

**Comment by Hyleus** [comment]
Yeah, this should be done. It would be ironic if Flathub, which is mostly composed of free software, to continue using a Git-hosting platform that is (now) owned by one of the biggest corporations and a big menace to free software/open-source projects.

I'd suggest Gitlab either directly or self-hosted, because it's FOSS and works well. Maybe the Gnome Gitlab instance could be considered?

**Comment by alexlarsson** [comment]
I don't think the microsoft news really changes anything. GitHub was always non-free, and we knew this from day one. The reason we chose GitHub was more because of that is what most people already had accounts on, and were used to workflows from.

The "production" side of flathub (i.e. shipping apps to users) is entierly independent on github, so switching to gitlab is not a huge technical issue. However, it will be a lot of pain with migrating issues, users, workflows, etc. I think a more proper reaction at this time is wait-and-see.

**Comment by alexlarsson** [comment]
It might be easy to make a copy of all things from github, but that doesn't mean every flathub maintainer has a user and we automatically map their permissions and update their workflows.

**Comment by Hyleus** [comment]
Ah that makes sense, I should have thought about it a bit more what problems might arise.... Well, good luck. Maybe it could be done like Gnome? Aren't they (the projects) given a choice there if they want to migrate to the new Gnome instance? Maybe it could be done partially, just an idea. At least it's seriously considered here!

I guess it makes sense to wait a bit more and to see how many people will switch to Gitlab.

**Comment by TingPing** [comment]
> Aren't they (the projects) given a choice there if they want to migrate to the new Gnome instance? 

No, everybody was moved over. Splitting all the repos between two just makes it harder on everybody trying to manage it.

**Comment by gnumdk** [comment]
>I guess it makes sense to wait a bit more and to see how many people will switch to Gitlab.

Ok, so I do the first move ;)
org.gnome.Eolie and org.gnome.Lollypop will be orphaned.

regards,

**Comment by Hyleus** [comment]
I guess it's time to also remove myself from Github. I hope I see you all on Gitlab or a self-hosted alternative.

**Comment by alexlarsson** [comment]
We should probably remove these then, I don't think having unmaintaned apps is a great idea.

**Comment by bilelmoussaoui** [comment]
@alexlarsson I can maintain both apps. I'm an active contributor to lollypop and I have contributed to eolie in the past.

**Comment by alexlarsson** [comment]
@bilelmoussaoui cool, i'll add you then

**Comment by alexlarsson** [comment]
@bilelmoussaoui Do you plan to do any updates of these? There has been releases, but no updates.

**Comment by bilelmoussaoui** [comment]
Oups! Will update both apps tonight. Will also subscribe to the Gitlab repos of both projects to make sure I receive the latest updates

**Comment by loupianche** [comment]
@bilelmoussaoui Updated on github but not on flathub :-/ There is something broken in flathub infra.

For the original issue, I really think flathub is going in the wrong direction with it centralized infrastructure.

We, developers, should be able to register a project on flathub, with any git provider.

**Comment by TingPing** [comment]
I completely disagree with that. All repos being controlled by Flathub means the git repo and their history can be trusted, I do not want to lose that just because some developer wants to host their own git repo.

**Comment by bilelmoussaoui** [comment]
@loupianche are you sure about that? both apps are up to date;)

**Comment by TingPing** [comment]
@bilelmoussaoui Build bot is failing today so the last commit didn't make it through. Anyway this is very off-topic.

**Comment by hadess** [comment]
> We, developers, should be able to register a project on flathub, with any git provider.

You can already package and serve your own applications, from your own infrastructure, if you are an upstream developer. I'm sure we'd find a way to point users to this upstream repository if you wanted to migrate away from Flathub.

**Comment by loupianche** [comment]
@hadess My main issue with flathub is that users think they are using last software release while most of the time flathub is lagging by many release.

> if you wanted to migrate away from Flathub.
The issue is that, as a developer, I have not time to follow all dependencies security issues and update my flatpak package. From this point of view, traditional package management is better.

**Comment by hadess** [comment]
> @hadess My main issue with flathub is that users think they are using last software release while most of the time flathub is lagging by many release.

That doesn't happen so often when upstream developers handle their Flathub repos. At least that's my case, where Flatpak building in Flathub is part of the release process.

> > if you wanted to migrate away from Flathub.
> 
> The issue is that, as a developer, I have not time to follow all dependencies security issues and update my flatpak package. From this point of view, traditional package management is better.

1) It doesn't matter as much if the application is sandboxed
2) You actually get to test the combination of libraries and software you ship inside the Flatpak, and not have things break from under you like in a distro.

In short, it's working as designed.

**Comment by loupianche** [comment]
>It doesn't matter as much if the application is sandboxed

Lollypop has filesystem access (users want to add files from a NFS mount point) so it does matter ;)

**Comment by bilelmoussaoui** [comment]
Like any other music player on Android too. Which has a better portals than flatpak/snap. Having a pop-up in the future to let the user decides whether they want to give it the permission to have home file access is great but it's an other discussion

**Comment by hadess** [comment]
> Lollypop has filesystem access (users want to add files from a NFS mount point) so it does matter ;)

Who said it doesn't?

Anyway, you not having the time to package for Flatpak (even if it eventually meant not having to maintain other distribution specific packages) is a very different problem to moving Flathub to another "git" site. I don't really see how moving Flathub to GitLab would give you more time, so this discussion is pretty pointless.

**Comment by TingPing** [comment]
Yes this is all off-topic, this is purely a question of migrating infrastructure and developers and has nothing to do with applications themselves.

**Comment by fbruetting** [comment]
Would be cool if GitLab coud be made to work somehow, maybe Flathub could expand to applications being providable either at GitHub or at GitLab (or at gitlab.gnome.org). As you can log in to GitLab with your GitHub account, I don’t really see a problem here.

For example there now is [Meteo](https://github.com/flathub/com.github.bitseater.weather), too, whos maintainer likes to provide a Flatpak and (totally understandable) doesn’t want to use GitHub, which is why that package now got EOL’d… This issue needs to be solved somehow… 😐

**Comment by amtlib-dot-dll** [comment]
I recommend this issue to be closed. Migrating the infrastructure leads to unpredictable errors all the time.

**Comment by lukas2005** [comment]
Btw gitlab is owned by Microsoft too https://twitter.com/EdgarSanchez/status/1004087778760646656

**Comment by maoschanz** [comment]
Doesn't this issue become relevant again, with the recent huge censorship issues?

**Comment by Mikaela** [comment]
> Doesn't this issue become relevant again, with the recent huge censorship problems?

For context, I guess you mean [Microsoft-Owned GitHub Blocks Devs in US Sanctioned Countries](https://www.bleepingcomputer.com/news/security/microsoft-owned-github-blocks-devs-in-us-sanctioned-countries/) or are there even more recent huge censorship problems?

**Comment by maoschanz** [comment]
It's this issue: flathub is not commercial so i don't think it's concerned by the sanctions, thus it should allow contributions from all countries

**Comment by BeatLink** [comment]
I really wish that Flathub could migrate to GitLab. If for nothing else, the ability to self host. Its pretty dangerous to have the entirety of the Flathub repo on a closed sourced walled garden platform that has previously demonstrated a willingness to take down open source software at the behest of corporate parties. Case in point, a few months back when Github took down Youtube DL after the RIAA filed a DCMA takedown notice: https://freedom.press/news/riaa-github-youtube-dl-journalist-tool/

What if Flathub hosted an open source software that a corporate giant didnt take too kindly to? What if said corporate giant asked Github to take flathub down? What will be done then? The migration will be difficult no doubt, but its worth it in the end to have your own hosting provider and platform that you can operate on your own terms.

**Comment by 12people** [comment]
There's another dimension to this that surprisingly doesn't seem to have been mentioned, and that's user privacy.

Flathub is effectively forcing Flatpak developers to create Github accounts, which means agreeing to Github's terms of service and privacy statement. Those give Microsoft the power to, among other things, track each visit, supplement your data with that from third parties, and also share your data with third parties.

While I'm intentionally choosing to compromise my user data here (and not just for the sake of Flathub), I don't think Flathub should require other developers to do the same. That's why a self-hosted Gitlab instance with strong privacy standards and no data-sharing with third parties seems to be the best solution here.

**Comment by TheEvilSkeleton** [comment]
Surprisingly, I'm pretty sure we don't need a GitHub account for Flathub. You can build outside of GitHub and use Flathub's credentials to simply publish, so theoretically your app can be on GitLab and use Flathub's credentials to publish.

Mozilla bundles Firefox in their own infra and publishes to Flathub, see: https://hg.mozilla.org/mozilla-central/file/tip/taskcluster/docker/firefox-flatpak. OBS Studio recently started doing that, see: https://github.com/obsproject/obs-studio/blob/d49c09c738f81a80ab40f02328d2079ea7084ac5/.github/workflows/flatpak.yml#L101-L114 and https://github.com/obsproject/obs-studio/pull/5569.

**Comment by 12people** [comment]
@TheEvilSkeleton Yes, that's true — Flathub requires making a Github account specifically for an app's metadata, not for the app's code.

The problem is that it still forces the developer to create a Github account — all the above privacy considerations still apply.

**Comment by razzeee** [comment]
https://github.com/flathub/website/pull/1516 would lift those requirements

**Comment by bbhtt** [comment]
There aren't any plans to migrate off of GitHub currently. The problems have been discussed in detail already.

Direct upload exists now and will be open to trusted entities only and handled on a case-by-case basis.

Anyone else is free to set up their Flatpak repo outside of Flathub if they are concerned about privacy. https://docs.flatpak.org/en/latest/publishing.html

**Comment by okias** [comment]
While I plan to keep my GitHub account, I noticed (unfortunately, not directly and with some delay) that some maintainers have left GitHub, resulting in their software being dropped from Flathub. Sadly, this information wasn’t propagated to GNOME Software, so I only discovered it much later, when I manually ran flatpak update. :cry:

This situation doesn’t feel right. I remember people using GitHub because it had the advantage of almost everyone having an account there, but it seems this trend is now reversing.

Personally, I believe it would be a great idea to ask Freedesktop to host Flathub. The Freedesktop GitLab instance is pure OSS, which aligns with the goal of keeping things accessible and open for everyone.

**Comment by bbhtt** [comment]
> Personally, I believe it would be a great idea to ask Freedesktop to host Flathub. he Freedesktop GitLab instance is pure OSS, 

Freedesktop Gitlab admins has made sufficiently clear in context of another unrelated migration that they want the gitlab instance to be used by projects that serve as "building blocks used by distributions". It makes little sense to ask them to host a bunch of application packaging recipe repos when that's not the goal there.

Additionally the FDO instance (or even the GNOME instance) places significant restrictions on new accounts to prevent spam, asking everyone to open an issue to get "additional" access. We often require new maintainers to be added to repos or first time developers submitting apps who never had anything to do with FDO gitlab. This would be a significant barrier to these activities. 

Gitlab also locks behind several features that we depend on, behind Enterprise/Ultimate and we have no chance of getting them.

Also this was mentioned previously, we currently have 3207 repos on GitHub and 2500+ external collaborators. Migrating all of those is no easy task and some can never be migrated. The perceived benefit from going "pure FOSS and privacy" would be lost if more than half of the catalogue of apps gets lost due to infra migration.

We won't be maintaining all our workflows, CI, actions, buildbot, flatpak-x-checker, linter, backends for two completely unrelated code hosts at no point and even more so when there are only <=6 people maintaining all this. That time is better spent improving current workflows.

**Comment by bbhtt** [comment]
I locked the thread since it is going over the same points again and again that was made in 2018.

**Comment by hfiguiere** [comment]
> some maintainers have left GitHub, resulting in their software being dropped from Flathub.

Let me rephrase this for you.

"some maintainers have left GitHub, resulting in abandoning the flathub package they submitted."

Flathub doesn't drop packages like that.
